lib/smooth/application/display/DisplaySpi.cpp

```diff
@@ -26,26 +26,25 @@ using namespace smooth::core::io::spi;
 
 namespace smooth::application::display
 {
-    static const char* TAG = "ILI9314";
+    static const char* TAG = "DisplaySpi";
     static const bool PIN_HIGH = true;
     static const bool PIN_LOW = false;
 
-    ILI9341::ILI9341(std::mutex& guard,
-                     gpio_num_t chip_select_pin,
-                     gpio_num_t data_command_pin,
-                     gpio_num_t reset_pin,
-                     gpio_num_t back_light_pin,
-                     uint8_t spi_command_bits,
-                     uint8_t spi_address_bits,
-                     uint8_t bits_between_address_and_data_phase,
-                     uint8_t spi_mode,
-                     uint8_t spi_positive_duty_cycle,
-                     uint8_t spi_cs_ena_posttrans,
-                     int spi_clock_speed_hz,
-                     uint32_t spi_device_flags,
-                     int spi_queue_size,
-                     bool use_pre_transaction_callback,
-                     bool use_post_transaction_callback)
+    DisplaySpi::DisplaySpi(std::mutex& guard,
+                           gpio_num_t chip_select_pin,
+                           gpio_num_t data_command_pin,
+                           gpio_num_t reset_pin,
+                           uint8_t spi_command_bits,
+                           uint8_t spi_address_bits,
+                           uint8_t bits_between_address_and_data_phase,
+                           uint8_t spi_mode,
+                           uint8_t spi_positive_duty_cycle,
+                           uint8_t spi_cs_ena_posttrans,
+                           int spi_clock_speed_hz,
+                           uint32_t spi_device_flags,
+                           int spi_queue_size,
+                           bool use_pre_transaction_callback,
+                           bool use_post_transaction_callback)
```

Comment:
An awful lot of parameters. When they get this many I prefer to create a configuration class that is passed to the constructor, esp. if most of them just are copied into the class in which case I just hold a copy of the configuration class instead. Not sure if it makes things easier here though.

Reply:
Almost all of these parameters get past to the base class SPIDevice.  Should we be looking at refractoring SPIDevice?

Reply:
Do you think it'll make it more readable? If yes, then that's an option.

Reply:
From my point of view I do not mind all the parameters and to me seeing everything that is required/needed to create the SPI device one place is the best approach.
``` c 
auto device = spi_master.create_device<DisplaySpi>(
                        GPIO_NUM_14,            // chip select gpio pin
                        GPIO_NUM_27,            // data command gpio pin
                        GPIO_NUM_33,            // reset gpio pin
                        0,                      // spi command_bits
                        0,                      // spi address_bits,
                        0,                      // bits_between_address_and_data_phase,
                        0,                      // spi_mode = 0,
                        128,                    // spi positive_duty_cycle,
                        0,                      // spi cs_ena_posttrans,
                        SPI_MASTER_FREQ_16M,    // spi-sck = 16MHz
                        0,                      // full duplex (4-wire)
                        7,                      // queue_size,
                        true,                   // use pre-trans callback
                        true);                  // use post-trans callback
```
I guess we could create a struct or class to hold the spi_config parameters but I don't feel it would make the code easier to read.

Reply:
Then leave it.